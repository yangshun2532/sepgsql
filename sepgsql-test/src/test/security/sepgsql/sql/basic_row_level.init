-- Basic Row-level Access Controls
-- ====================================

SELECT sepgsql_getcon();

-- cleanup previous tests
SET client_min_messages TO 'error';

DROP TABLE IF EXISTS t1;

RESET client_min_messages;

CREATE TABLE t1
(
	a	int primary key,
	b	text
);

INSERT INTO t1(security_label, a, b) VALUES
       ('system_u:object_r:sepgsql_table_t:s0',            1, 'aaa'),
       ('system_u:object_r:sepgsql_table_t:s0:c0',         2, 'bbb'),
       ('system_u:object_r:sepgsql_table_t:s0:c1',         3, 'ccc'),
       ('system_u:object_r:sepgsql_ro_table_t:s0',         4, 'ddd'),
       ('system_u:object_r:sepgsql_ro_table_t:s0:c0',      5, 'eee'),
       ('system_u:object_r:sepgsql_ro_table_t:s0:c1',      6, 'fff'),
       ('system_u:object_r:sepgsql_fixed_table_t:s0',      7, 'ggg'),
       ('system_u:object_r:sepgsql_fixed_table_t:s0:c0',   8, 'hhh'),
       ('system_u:object_r:sepgsql_fixed_table_t:s0:c1',   9, 'iii'),
       ('system_u:object_r:sepgsql_secret_table_t:s0',    10, 'aaa'),
       ('system_u:object_r:sepgsql_secret_table_t:s0:c0', 11, 'aaa'),
       ('system_u:object_r:sepgsql_secret_table_t:s0:c1', 12, 'aaa');
