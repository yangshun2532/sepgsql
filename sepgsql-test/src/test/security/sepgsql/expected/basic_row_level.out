-- Basic Row-level Access Controls
-- ====================================
SELECT sepgsql_getcon();
                   sepgsql_getcon                    
-----------------------------------------------------
 unconfined_u:unconfined_r:unconfined_t:s0-s0:c0.c15
(1 row)

-- cleanup previous tests
SET client_min_messages TO 'error';
DROP TABLE IF EXISTS t1;
RESET client_min_messages;
CREATE TABLE t1
(
	a	int primary key,
	b	text
);
psql:sql/basic_row_level.init:17: NOTICE:  CREATE TABLE / PRIMARY KEY will create implicit index "t1_pkey" for table "t1"
INSERT INTO t1(security_label, a, b) VALUES
       ('system_u:object_r:sepgsql_table_t:s0',            1, 'aaa'),
       ('system_u:object_r:sepgsql_table_t:s0:c0',         2, 'bbb'),
       ('system_u:object_r:sepgsql_table_t:s0:c1',         3, 'ccc'),
       ('system_u:object_r:sepgsql_ro_table_t:s0',         4, 'ddd'),
       ('system_u:object_r:sepgsql_ro_table_t:s0:c0',      5, 'eee'),
       ('system_u:object_r:sepgsql_ro_table_t:s0:c1',      6, 'fff'),
       ('system_u:object_r:sepgsql_fixed_table_t:s0',      7, 'ggg'),
       ('system_u:object_r:sepgsql_fixed_table_t:s0:c0',   8, 'hhh'),
       ('system_u:object_r:sepgsql_fixed_table_t:s0:c1',   9, 'iii'),
       ('system_u:object_r:sepgsql_secret_table_t:s0',    10, 'aaa'),
       ('system_u:object_r:sepgsql_secret_table_t:s0:c0', 11, 'aaa'),
       ('system_u:object_r:sepgsql_secret_table_t:s0:c1', 12, 'aaa');
-- Basic Row-level Access Controls
-- ====================================
SELECT sepgsql_getcon();
                  sepgsql_getcon                   
---------------------------------------------------
 unconfined_u:unconfined_r:sepgsql_test_t:s0-s0:c0
(1 row)

SELECT security_label, * FROM t1;
                security_label                 | a |  b  
-----------------------------------------------+---+-----
 system_u:object_r:sepgsql_table_t:s0          | 1 | aaa
 system_u:object_r:sepgsql_table_t:s0:c0       | 2 | bbb
 system_u:object_r:sepgsql_ro_table_t:s0       | 4 | ddd
 system_u:object_r:sepgsql_ro_table_t:s0:c0    | 5 | eee
 system_u:object_r:sepgsql_fixed_table_t:s0    | 7 | ggg
 system_u:object_r:sepgsql_fixed_table_t:s0:c0 | 8 | hhh
(6 rows)

COPY t1 (security_label, a, b) TO stdout;
system_u:object_r:sepgsql_table_t:s0	1	aaa
system_u:object_r:sepgsql_table_t:s0:c0	2	bbb
system_u:object_r:sepgsql_ro_table_t:s0	4	ddd
system_u:object_r:sepgsql_ro_table_t:s0:c0	5	eee
system_u:object_r:sepgsql_fixed_table_t:s0	7	ggg
system_u:object_r:sepgsql_fixed_table_t:s0:c0	8	hhh
BEGIN;
UPDATE t1 SET b = b || '_updt';
SELECT security_label, * FROM t1;
                security_label                 | a |    b     
-----------------------------------------------+---+----------
 system_u:object_r:sepgsql_ro_table_t:s0       | 4 | ddd
 system_u:object_r:sepgsql_ro_table_t:s0:c0    | 5 | eee
 system_u:object_r:sepgsql_fixed_table_t:s0    | 7 | ggg
 system_u:object_r:sepgsql_fixed_table_t:s0:c0 | 8 | hhh
 system_u:object_r:sepgsql_table_t:s0          | 1 | aaa_updt
 system_u:object_r:sepgsql_table_t:s0:c0       | 2 | bbb_updt
(6 rows)

ABORT;
BEGIN;
DELETE FROM t1;
SELECT security_label, * FROM t1;
                security_label                 | a |  b  
-----------------------------------------------+---+-----
 system_u:object_r:sepgsql_ro_table_t:s0       | 4 | ddd
 system_u:object_r:sepgsql_ro_table_t:s0:c0    | 5 | eee
 system_u:object_r:sepgsql_fixed_table_t:s0    | 7 | ggg
 system_u:object_r:sepgsql_fixed_table_t:s0:c0 | 8 | hhh
(4 rows)

ABORT;
